php/ad-servers/class-ad-layers-dfp.php

```diff
@@ -551,7 +551,9 @@ private function ad_unit_js( $ad_layer ) {
 					$is_oop ? '' : ',' . wp_json_encode( $this->default_by_unit[ $ad_unit ] ),
 					wp_json_encode( $this->get_ad_unit_id( $ad_unit ) ),
 					( ! empty( $this->mapping_by_unit[ $ad_unit ] ) && ! in_array( $ad_unit, $this->oop_units ) ) ? '.defineSizeMapping(dfpBuiltMappings[' . wp_json_encode( $ad_unit ) . '])' : '',
-					( ! empty( $this->targeting_by_unit[ $ad_unit ] ) ) ? $this->targeting_by_unit[ $ad_unit ] : '' // This is escaped above as it is built
+					( ! empty( $this->targeting_by_unit[ $ad_unit ] ) ) ? $this->targeting_by_unit[ $ad_unit ] : '', // This is escaped above as it is built
+					// allow addition services such as '.setCollapseEmptyDiv(true,true)' to be added to the define call
+					apply_filters( 'ad_layers_dfp_additional_services', $ad_unit )
```

Comment:
By default, this will add the value of `$ad_unit` to the JS chain, which will almost always cause a syntax error. I think you probably want to call `apply_filters( 'ad_layers_dfp_additional_services', '', $ad_unit )`, so by default it will drop in an empty string.